Dockerfile

```diff
@@ -1,86 +1,74 @@
 
+# Stage to download the gRPC health probe
 FROM public.ecr.aws/docker/library/alpine:latest AS binaries
 
-# Install packages needed to build binaries
-RUN apk update --no-cache && \
-    apk upgrade --no-cache && \
-    apk add --update --no-cache wget curl jq
-
-RUN [[ "$(arch)" = "aarch64" ]] && arch="arm64" || arch="$(arch)" && \
-  curl -s https://api.github.com/repos/grpc-ecosystem/grpc-health-probe/releases/latest \
-  | jq ".assets[].browser_download_url" \
-  | grep "grpc_health_probe-linux-$arch" \
-  | tr -d '"' \
-  | wget -qi - -O grpc-health-probe && \
-  chmod +x grpc-health-probe && \
-  mv grpc-health-probe /usr/local/bin
-
-# Throw-away build stage to reduce size of final image
+# Install packages to download the gRPC health probe
+RUN apk add --no-cache wget curl jq
+
+# Download and configure the gRPC health probe
+RUN ARCH=$(uname -m) && \
+    [ "$ARCH" = "x86_64" ] && ARCH="amd64" || ARCH="$ARCH" && \
+    wget -q -O /usr/local/bin/grpc-health-probe \
+      $(curl -s https://api.github.com/repos/grpc-ecosystem/grpc-health-probe/releases/latest \
+      | jq -r ".assets[] | select(.name | test(\"linux-$ARCH\")) | .browser_download_url") && \
+    chmod +x /usr/local/bin/grpc-health-probe
+
+# Build stage
 FROM base as build
 
-# Install packages needed to build gems
-RUN apk update --no-cache && \
-    apk upgrade --no-cache && \
-    apk add --update --no-cache build-base git gcompat postgresql-dev nodejs yarn
+# Install build packages and temporary dependencies
+RUN apk add --no-cache --virtual .build-deps build-base git gcompat postgresql-dev nodejs yarn
 
 # Install application gems
 COPY Gemfile Gemfile.lock ./
-
 RUN bundle install && \
     rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
-    find "${BUNDLE_PATH}" -name "*.o" -delete && find "${BUNDLE_PATH}" -name "*.c" -delete && \
-    mkdir -p tmp/pids && \
-    mkdir -p storage && \
-    bundle exec bootsnap precompile --gemfile
+    find "${BUNDLE_PATH}" -name "*.o" -delete && \
+    find "${BUNDLE_PATH}" -name "*.c" -delete
 
+# Install Node.js packages and dependencies
 COPY package.json yarn.lock ./
 RUN yarn install --frozen-lockfile
 
-# Copy application code
+# Copy and precompile application code
 COPY . .
+RUN bundle exec bootsnap precompile --gemfile && \
+    SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile
 
-# Precompile bootsnap code for faster boot times
-RUN bundle exec bootsnap precompile app/ lib/
-
-# Precompiling assets for production without requiring secret RAILS_MASTER_KEY
-RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile
-
-# Final stage for app image
+# Final production stage
 FROM base
 
-# Install packages needed for deployment
-RUN apk update --no-cache && \
-    apk upgrade --no-cache && \
-    apk add --update --no-cache build-base gcompat postgresql-dev vips-dev ffmpeg
+# Install only runtime dependencies
+RUN apk add --no-cache postgresql-libs vips-dev ffmpeg
```

Comment:
I'm can't remember why, and I should have left a better description but this commit: https://github.com/somleng/somleng/pull/1721/checks?check_run_id=32366076143 adds gcompat. I recall it had something to do with alpine arm64 builds. Is there a reason why you removed this?

Reply:
This item may be rolled back, I will review this item as well.